openhcl/page_pool_alloc/src/lib.rs

```diff
@@ -23,6 +23,261 @@ use std::sync::Arc;
 use thiserror::Error;
 use vm_topology::memory::MemoryRangeWithNode;
 
+/// Save restore suport for [`PagePool`].
+pub mod save_restore {
+    use super::DeviceId;
+    use super::PagePool;
+    use super::State;
+    use memory_range::MemoryRange;
+    use mesh::payload::Protobuf;
+    use vmcore::save_restore::SaveRestore;
+    use vmcore::save_restore::SavedStateRoot;
+
+    #[derive(Protobuf)]
+    #[mesh(package = "openhcl.pagepool")]
+    enum InnerState {
```

Comment:
I'd suggest lifting the common fields out of each variant and make this a struct containing the state enum.